gio/src/subclass/vfs.rs

```diff
@@ -0,0 +1,184 @@
+
+// Support custom implementation of virtual functions defined in `gio::ffi::GVfsClass`.
+pub trait VfsImpl: ObjectImpl + ObjectSubclass<Type: IsA<Vfs>> {
+    fn is_active(&self) -> bool {
+        self.parent_is_active()
+    }
+
+    fn get_file_for_path(&self, path: impl AsRef<std::path::Path>) -> File {
+        self.parent_get_file_for_path(path)
+    }
+
+    fn get_file_for_uri(&self, uri: &str) -> File {
+        self.parent_get_file_for_uri(uri)
+    }
+
+    fn get_supported_uri_schemes(&self) -> Vec<String> {
+        self.parent_get_supported_uri_schemes()
+    }
+
+    fn parse_name(&self, parse_name: &str) -> File {
+        self.parent_parse_name(parse_name)
+    }
+}
+
+// Support parent implementation of virtual functions defined in `gio::ffi::GVfsClass`.
+pub trait VfsImplExt: VfsImpl {
+    fn parent_is_active(&self) -> bool {
+        unsafe {
+            let data = Self::type_data();
+            let parent_class = data.as_ref().parent_class() as *const ffi::GVfsClass;
+
+            let f = (*parent_class)
+                .is_active
+                .expect("No parent class implementation for \"is_active\"");
+
+            let res = f(self.obj().unsafe_cast_ref::<Vfs>().to_glib_none().0);
+            from_glib(res)
+        }
+    }
+
+    fn parent_get_file_for_path(&self, path: impl AsRef<std::path::Path>) -> File {
+        unsafe {
+            let data = Self::type_data();
+            let parent_class = data.as_ref().parent_class() as *const ffi::GVfsClass;
+
+            let f = (*parent_class)
+                .get_file_for_path
+                .expect("No parent class implementation for \"get_file_for_path\"");
+
+            let res = f(
+                self.obj().unsafe_cast_ref::<Vfs>().to_glib_none().0,
+                path.as_ref().to_glib_none().0,
+            );
+            from_glib_full(res)
+        }
+    }
+
+    fn parent_get_file_for_uri(&self, uri: &str) -> File {
+        unsafe {
+            let data = Self::type_data();
+            let parent_class = data.as_ref().parent_class() as *const ffi::GVfsClass;
+
+            let f = (*parent_class)
+                .get_file_for_uri
+                .expect("No parent class implementation for \"get_file_for_uri\"");
+
+            let res = f(
+                self.obj().unsafe_cast_ref::<Vfs>().to_glib_none().0,
+                uri.to_glib_none().0,
+            );
+            from_glib_full(res)
+        }
+    }
+
+    fn parent_get_supported_uri_schemes(&self) -> Vec<String> {
+        unsafe {
+            let data = Self::type_data();
+            let parent_class = data.as_ref().parent_class() as *const ffi::GVfsClass;
+
+            let f = (*parent_class)
+                .get_supported_uri_schemes
+                .expect("No parent class implementation for \"get_supported_uri_schemes\"");
+
+            let res = f(self.obj().unsafe_cast_ref::<Vfs>().to_glib_none().0);
+            FromGlibPtrContainer::from_glib_none(res)
```

Comment:
This is suboptimal as it copies strings all the time. Maybe this should use a `&glib::StrV`?

Reply:
`parent_get_supported_uri_schemes()` will already return a `&'static StrV`, you can just pass it further without any copying or anything else.

Actually, I see your problem. Yes, this actually has to return a `&'static [glib::GStringPtr]`. You can get that from `StrV::from_glib_borrow()` for example. User implementations can build a `&'static StrV` (e.g. via `LazyLock`) and that derefs into `&'static [glib::GStringPtr]`

Reply:
Indeed. Maybe a reason to make it `&'static StrV`. You can get a stable pointer out of that :)